.github/workflows/ci.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node_version: [12, 14, 15]
+        include:
+          - os: macos-latest
+            node_version: 14
+          - os: windows-latest
+            node_version: 14
+      fail-fast: false
+
+    name: 'Build&Test: node-${{ matrix.node_version }}, ${{ matrix.os }}'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set node version to ${{ matrix.node_version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node_version }}
+
+      - name: Versions
+        run: yarn versions
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Build vite
+        run: yarn build-vite
+
+      - name: Build plugin-vue
+        run: yarn build-plugin-vue
+
+      - name: Test serve
+        run: yarn test-serve --runInBand
+
+      - name: Test build
+        run: yarn test-build --runInBand
+
+  lint:
```

Comment:
Do we need to have this separately? The installing/build processes are kinda duplicated to me, maybe we can have it inside the `build` job with [`job.continue-on-error`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) to have both lint errors and test errors, and also [`job.if`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif) make it only run once.

Reply:
You can shrink this if you want.

```yaml
    runs-on: ubuntu-latest
    name: 'Lint'
    steps:
      - uses: actions/checkout@v2
      - name: Prepare
        run: |
          yarn install --frozen-lockfile
          yarn build-vite
          yarn build-plugin-vue
      - run: yarn lint
```

I would put those yarn install and build stuff into the `prepare` so they are run with `yarn install`

Reply:
@aminya 

I personally like it more clean and all have set explicit `name`  
But if really wanted I can omit names here and there

Also I want to set node to version 14 explicitly, so that if `actions/setup-node@v2` / GitHub Actions would get an update, we have the control of what to update when.

---

@antfu 

Having it separately makes a bit sense here, cause the job itself is specialized for doing one thing and that also quickly

Separating the jobs like this also helps a bit to jump into e.g. special test errors. So not everything is made at one.